pkg/apis/certmanager/v1alpha1/types_issuer.go

```diff
@@ -204,6 +205,15 @@ type ACMEIssuerDNS01ProviderAcmeDNS struct {
 	AccountSecret SecretKeySelector `json:"accountSecretRef"`
 }
 
+// ACMEIssuerDNS01ProviderDNSMadeEasy is a structure containing the
+// configuration for DNSMadeEasy
+type ACMEIssuerDNS01ProviderDNSMadeEasy struct {
+	BaseURL   string            `json:"baseURL"`
+	APIKey    string            `json:"apiKey"`
+	SecretKey SecretKeySelector `json:"secretKeySecretRef"`
+	DomainID  uint              `json:"domainID"`
```

Comment:
Please add comments to each of these fields, as they will get pulled into the documentation website automatically.

Reply:
Also, I think adding `// +optional` tags to those fields which are not required would be ideal.

Reply:
++ we don't use `uint` in our API types (if i recall correctly), as per k8s conventions.

Please change this to an `int` and add a validation check to ensure the domain ID is >=0

Reply:
Roger, I'll make these changes over the weekend.